cmd/modelcmd/apicontext.go

```diff
@@ -17,11 +19,19 @@ import (
 	"github.com/juju/juju/jujuclient"
 )
 
+type CookieJar interface {
+	http.CookieJar
+	RemoveAllHost(host string)
+}
+
 // APIContext holds the context required for making connections to
 // APIs used by juju.
 type APIContext struct {
-	Jar            *cookiejar.Jar
-	WebPageVisitor httpbakery.Visitor
+	Jar CookieJar
+	// jar holds the internal version of the cookie jar - it has
+	// methods that clients should not use, such as Save.
+	jar            *domainCookieJar
```

Comment:
They're actually the same object. The exported one is an interface type so that we can hide methods that we don't want external packages to invoke. Because they're the same object, it doesn't really matter which one is used, so ISTM that using a similar name is OK.

I could rename jar to internalJar if you'd prefer.

> "we send it to everyone because we don't know who needs it", feels very much like a security hole.

Can you think of any actual security hole that might result from sending our preferred authentication domain to the parties that we make HTTP requests to?

I guess it might be possible to add a hook to the bakery package so that we could add the cookie only when discharging, but at the moment that seems like more complexity than is warranted.